test/srt/test_fused_moe.py

```diff
@@ -0,0 +1,97 @@
+import unittest
+import torch
+from sglang.srt.layers.activation import SiluAndMul
+from sglang.srt.layers.fused_moe_triton.fused_moe import fused_moe
+from vllm.model_executor.layers.fused_moe import fused_moe as fused_moe_vllm
+
+class TestFusedMOE(unittest.TestCase):
+    NUM_EXPERTS = [8, 64]
+    TOP_KS = [2, 6]
+    
+    def torch_naive_moe(self, a, w1, w2, score, topk):
+        B, D = a.shape
+        a = a.view(B, -1, D).repeat(1, topk, 1).reshape(-1, D)
+        out = torch.zeros(B * topk, w2.shape[1], dtype=a.dtype, device=a.device)
+        score = torch.softmax(score, dim=-1, dtype=torch.float32)
+        topk_weight, topk_ids = torch.topk(score, topk)
+        topk_weight = topk_weight.view(-1)
+        topk_ids = topk_ids.view(-1)
+        for i in range(w1.shape[0]):
+            mask = topk_ids == i
+            if mask.sum():
+                out[mask] = SiluAndMul()(
+                    a[mask] @ w1[i].transpose(0, 1)) @ w2[i].transpose(0, 1)
+        return (out.view(B, -1, w2.shape[1]) *
+                topk_weight.view(B, -1, 1).to(out.dtype)).sum(dim=1)
+
+    def _test_case(self, m, n, k, e, topk, dtype, use_fp8_w8a8=False):
+        if use_fp8_w8a8:
+            # AssertionError: fp8e4nv data type is not supported on CUDA arch < 89
```

Comment:
If the GPU is Ampere architecture, we should fallback from `fused_moe` to either naive implementation or `torch.compile` implementation to prevent errors.